pkg/cmd/cli/cmd/projects.go

```diff
@@ -0,0 +1,182 @@
+	kubecmdconfig "k8s.io/kubernetes/pkg/kubectl/cmd/config"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/client"
+	cliconfig "github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/project/api"
+
+	"github.com/spf13/cobra"
+)
+
+type ProjectsOptions struct {
+	Config       clientcmdapi.Config
+	ClientConfig *restclient.Config
+	ClientFn     func() (*client.Client, error)
+	Out          io.Writer
+	PathOptions  *kubecmdconfig.PathOptions
+
+	ProjectOnly  bool
+	DisplayShort bool
+
+	// SkipAccessValidation means that if a specific name is requested, don't bother checking for access to the project
+	SkipAccessValidation bool
+}
+
+const (
+	projectsLong = `
+Display information about the current active project and existing projects on the server.
+
+For advanced configuration, or to manage the contents of your config file, use the 'config'
+command.`
+
+	projectsExample = `  # Display the projects that currently exist
+  %[1]s`
+)
+
+// NewCmdProjects implements the OpenShift cli rollback command
+func NewCmdProjects(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &ProjectsOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "projects",
+		Short:   "Display existing projects",
+		Long:    projectsLong,
+		Example: fmt.Sprintf(projectsExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			options.PathOptions = cliconfig.NewPathOptions(cmd)
+
+			if err := options.Complete(f, args, out); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			if err := options.RunProjects(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+	return cmd
+}
+
+func (o *ProjectsOptions) Complete(f *clientcmd.Factory, args []string, out io.Writer) error {
+	var err error
+
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments should be passed")
+	}
+
+	o.Config, err = f.OpenShiftClientConfig.RawConfig()
+	if err != nil {
+		return err
+	}
+
+	o.ClientConfig, err = f.OpenShiftClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	o.ClientFn = func() (*client.Client, error) {
+		client, _, err := f.Clients()
+		return client, err
+	}
+
+	o.Out = out
+
+	return nil
+}
+func (o ProjectsOptions) Validate() error {
+	return nil
+}
+
+// RunProjects lists all projects a user belongs to
+func (o ProjectsOptions) RunProjects() error {
+	config := o.Config
+	clientCfg := o.ClientConfig
+	out := o.Out
+
+	currentContext := config.Contexts[config.CurrentContext]
+	currentProject := currentContext.Namespace
+
+	var currentProjectExists bool = false
+
+	client, err := o.ClientFn()
+	if err != nil {
+		return err
+	}
+
+	if len(currentProject) > 0 {
+		if o.DisplayShort {
+			fmt.Fprintln(out, currentProject)
+			return nil
+		}
+
+		if _, err := client.Projects().Get(currentProject); err != nil {
+			if kapierrors.IsNotFound(err) {
```

Comment:
I'm not sure if we should fail in these cases. That's correct for `oc project` where the main purpose is to display the current project, but here the major path is to list projects. What do you think about using the same form of output, just not display the current one, maybe with the error and some guidance at the bottom?

```
$ oc projects
You have access to the following projects and can switch between them with 'oc project <projectname>':

  * awesome
  * notsoawesome
  * okone
  * weirdo

You do not have rights to view the project "myinvalid" specified in your config. Please switch to an existing one.
```